bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/api/DeviceController.java

```diff
@@ -0,0 +1,205 @@
+                APIdata.static_access_token);
+        Gson gson = new Gson();
+        this.realTimeBattery = gson.fromJson(response, Battery.class);
+    }
+
+    private void getInverterACDCTemperatures() throws IOException, JsonSyntaxException {
+        String date = getAPIFormatDate();
+        String response = apiGetMethod(
+                makeURL("api/v1/inverter/" + this.sn + "/output/day?lan=en&date=" + date + "&column=dc_temp,igbt_temp"),
+                APIdata.static_access_token);
+        Gson gson = new Gson();
+        this.inverter_day_temperatures = gson.fromJson(response, Daytemps.class);
+        this.inverter_day_temperatures.getLastValue();
+    }
+
+    private void getRealTimeIn() throws IOException, JsonSyntaxException { // Get URL Respnse
+        String response = apiGetMethod(makeURL("api/v1/inverter/" + this.sn + "/realtime/input"),
+                APIdata.static_access_token);
+        Gson gson = new Gson();
+        this.realTimeDataIn = gson.fromJson(response, RealTimeInData.class);
+        this.realTimeDataIn.sumPVIV();
+    }
+
+    public void sendSettings(@Nullable Settings settings) throws SunSynkSendCommandException {
+        String body = settings.buildBody();
+        sendCommandToSunSynk(body);
+    }
+
+    public void sendCommandToSunSynk(String body) throws SunSynkSendCommandException {
+        String path = "api/v1/common/setting/" + this.sn + "/set";
+
+        try {
+            apiPostMethod(makeURL(path), body, APIdata.static_access_token);
+        } catch (IOException e) {
+            logger.debug("Failed to send to Inverter API: {} ", e.getMessage());
+            int found = e.getMessage().indexOf("Authentication challenge without WWW-Authenticate header");
+            if (found > -1) {
+                throw new SunSynkSendCommandException("Authentication token failed", e);
+            }
+            throw new SunSynkSendCommandException("Unknown athentication fail", e);
+        }
+    }
+
+    private String apiPostMethod(String httpsURL, String body, String access_token) throws IOException {
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/json");
+        headers.setProperty("Authorization", "Bearer " + access_token);
+        headers.setProperty("Requester", "www.openhab.org"); // optional
+        InputStream stream = new ByteArrayInputStream(body.getBytes(StandardCharsets.UTF_8));
+        return HttpUtil.executeUrl("POST", httpsURL, headers, stream, "application/json", 2000);
+    }
+
+    private String apiGetMethod(String httpsURL, String access_token) throws IOException {
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/json");
+        headers.setProperty("Requester", "www.openhab.org"); // optional
+        headers.setProperty("Authorization", "Bearer " + access_token);
+        return HttpUtil.executeUrl("GET", httpsURL, headers, null, "application/json", 2000);
```

Comment:
1. I would prefer to use the [Jetty HttpMethod enum](https://javadoc.jetty.org/jetty-12/org/eclipse/jetty/http/HttpMethod.html) instead of magic strings like "GET"
2. I would prefer to extract the timeout to class level constant. And 2 seconds seems a bit short.
This applies to all executeUrl calls in this class.

Reply:
okay, also done in AccountController.java